GitUI/GitExtensionsDialog.cs

```diff
@@ -31,5 +32,42 @@ protected GitExtensionsDialog([NotNull] GitUICommands commands, bool enablePosit
             // Lighten up the control panel
             ControlsPanel.BackColor = KnownColor.ControlLight.MakeBackgroundDarkerBy(-0.04);
         }
+
+        /// <summary>
+        /// Gets or sets the anchor pointing to a section in the manual pertaining to this dialog.
+        /// </summary>
+        /// <remarks>
+        /// The URL structure:
+        /// https://git-extensions-documentation.readthedocs.io/en/latest/{ManualSectionSubfolder}.html#{ManualSectionAnchorName}
+        /// </remarks>
+        public string ManualSectionAnchorName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of a document pertaining to this dialog.
+        /// </summary>
+        /// <remarks>
+        /// The URL structure:
+        /// https://git-extensions-documentation.readthedocs.io/en/latest/{ManualSectionSubfolder}.html#{ManualSectionAnchorName}
+        /// </remarks>
+        public string ManualSectionSubfolder { get; set; }
+
+        protected override void OnHelpButtonClicked(CancelEventArgs e)
+        {
+            // If we show the Help button but we have failed to specify where the docs are -> hide the button, and exit
+            if (string.IsNullOrWhiteSpace(ManualSectionAnchorName) || string.IsNullOrWhiteSpace(ManualSectionSubfolder))
+            {
+                HelpButton = false;
```

Comment:
Hide on click feels like a little late.

Reply:
This is more a backup strategy if the help is misconfigured.